File: FHSSpirit/FHSSpiritDataModelsFS/ItemMessageCommentModel.fs
namespace FHSSpiritDataModelsFS

open System.ComponentModel

type ItemMessageCommentModel() =  
    let mutable m_nID : int = 0
    let mutable m_strCreationDateTime : string = ""
    let mutable m_strContent : string = ""
    let mutable m_strOwner : string = ""
    let mutable str : string = ""

    let ev = new Event<_,_>()
    interface INotifyPropertyChanged with
        [<CLIEvent>]
        member x.PropertyChanged = ev.Publish

    member x.ID
        with get() = m_nID
        and set(id') =
                m_nID <- id'
                ev.Trigger(x, PropertyChangedEventArgs("ID"))

    member x.CreationDateTime
        with get() = m_strCreationDateTime
        and set(creationDateTime') =
                m_strCreationDateTime <- creationDateTime'
                ev.Trigger(x, PropertyChangedEventArgs("CreationDateTime"))

    member x.Content
        with get() = m_strContent
        and set(content') =
                m_strContent <- content'
                ev.Trigger(x, PropertyChangedEventArgs("Content"))
                
    member x.Owner
        with get() = m_strOwner
        and set(owner') =
                m_strOwner <- owner'
                ev.Trigger(x, PropertyChangedEventArgs("Owner"))

    member x.StringProp
        with get() = str
            and set(str') =
                str <- str'
                ev.Trigger(x, PropertyChangedEventArgs("StringProp"))

File: FHSSpirit/FHSSpiritDataModels/ItemMessageCommentModel.cs
using System;
using System.ComponentModel;
using System.Diagnostics;
using System.Net;
using System.Windows;
using System.Windows.Controls;
using System.Windows.Documents;
using System.Windows.Ink;
using System.Windows.Input;
using System.Windows.Media;
using System.Windows.Media.Animation;
using System.Windows.Shapes;

namespace FHSSpiritDataModels
{
    /*public class ItemMessageCommentModel : INotifyPropertyChanged
    {
        /// <summary>
        /// Modell für News
        /// </summary>
        private int m_nID;
        /// <summary>
        /// ViewModel-Beispieleigenschaft. Diese Eigenschaft wird in der Ansicht verwendet, um den Wert unter Verwendung einer Bindung anzuzeigen.
        /// </summary>
        /// <returns></returns>
        public int ID
        {
            get
            {
                return m_nID;
            }
            set
            {
                if (value != m_nID)
                {
                    m_nID = value;
                    NotifyPropertyChanged("ID");
                }
            }
        }

        private string m_strCreationDateTime;
        /// <summary>
        /// ViewModel-Beispieleigenschaft. Diese Eigenschaft wird in der Ansicht verwendet, um den Wert unter Verwendung einer Bindung anzuzeigen.
        /// </summary>
        /// <returns></returns>
        public string CreationDateTime
        {
            get
            {
                return m_strCreationDateTime;
            }
            set
            {
                if (value != m_strCreationDateTime)
                {
                    m_strCreationDateTime = value;
                    NotifyPropertyChanged("ID");
                }
            }
        }

        private string m_strContent;
        /// <summary>
        /// ViewModel-Beispieleigenschaft. Diese Eigenschaft wird in der Ansicht verwendet, um den Wert unter Verwendung einer Bindung anzuzeigen.
        /// </summary>
        /// <returns></returns>
        public string Content
        {
            get
            {
                return m_strContent;
            }
            set
            {
                if (value != m_strContent)
                {
                    m_strContent = value;
                    NotifyPropertyChanged("Content");
                }
            }
        }

        private string m_strOwner;
        /// <summary>
        /// ViewModel-Beispieleigenschaft. Diese Eigenschaft wird in der Ansicht verwendet, um den Wert unter Verwendung einer Bindung anzuzeigen.
        /// </summary>
        /// <returns></returns>
        public string Owner
        {
            get
            {
                return m_strOwner;
            }
            set
            {
                if (value != m_strOwner)
                {
                    m_strOwner = value;
                    NotifyPropertyChanged("Owner");
                }
            }
        }

        public event PropertyChangedEventHandler PropertyChanged;
        private void NotifyPropertyChanged(String propertyName)
        {
            PropertyChangedEventHandler handler = PropertyChanged;
            if (null != handler)
            {
                handler(this, new PropertyChangedEventArgs(propertyName));
            }
        }
    }*/
}

